bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/MessageBox.java

```diff
@@ -99,6 +106,88 @@ public MessageBox (Shell parent) {
 public MessageBox (Shell parent, int style) {
 	super (parent, checkStyle (parent, checkStyle (style)));
 	checkSubclass ();
+	/* Create the callback hook */
+	int threadId = OS.GetCurrentThreadId ();
+	Callback cbtCallback = new Callback (this, "CBTProc", 3); //$NON-NLS-1$
+	cbtHook = OS.SetWindowsHookEx (OS.WH_CBT, cbtCallback.getAddress (), 0, threadId);
+}
+
+long CBTProc(long nCode, long wParam, long lParam) {
+	char[] buffer = new char[128];
+	int length = OS.GetClassName(wParam, buffer, buffer.length);
+	String className = new String(buffer, 0, length);
+	if (labels != null) {
+		switch ((int) nCode) {
+		/* Cache the button handles in the order of creation */
```

Comment:
Use `GetDlgItem()` instead. See this for an example:
https://www.codeguru.com/windows/messagebox-with-custom-button-captions/

Reply:
Done.